src/main/kotlin/org/rust/cargo/project/workspace/CargoWorkspace.kt

```diff
@@ -234,9 +248,8 @@ private class WorkspaceImpl(
                         val dependencyPackage = newIdToPackage[pkg.id] ?: return@mapNotNull null
                         DependencyImpl(dependencyPackage, name, depKinds)
                     })
-                    pkg.dependencies.addAll(stdlibDependencies.filter { it.pkg.id in stdRoots })
-                    val explicitDeps = pkg.dependencies.map { it.pkg.id }.toSet()
-                    pkg.dependencies.addAll(stdlibDependencies.filter { it.pkg.id in stdGated && it.pkg.id !in explicitDeps })
+                    val explicitDeps = pkg.dependencies.map { it.name }.toSet()
+                    pkg.dependencies.addAll(stdlibDependencies.filter { it.name !in explicitDeps })
```

Comment:
Here I allowed overriding any stdlib dependency. E.g. in `rustc` it's common to write
```
[dependencies]
core = { path = "../libcore" }
```
So, we should not add one more "core" in this case